src/SQL.php

```diff
@@ -15,6 +15,13 @@
 use FluentSql\Functions\SqlFunction;
 
 class SQL {
+
+	/** @const FORCE_NON_NUMERIC_LIMIT whether to force using non numeric LIMIT clause, e.g. a subquery */
```

Comment:
Oh, this might be an answer to the question above. Are we doing this anywhere?

I would argue that the safety is likely more important than the flexibility in this case. The efficiency gains from deriving the `LIMIT` or `OFFSET` from a sub query are probably minimal since you need a single numerical value which you could easily do in a separate query+response with minimal overhead.